frontend/src/metabase/visualizations/components/settings/ChartSettingsListColumns/ChartSettingsListColumns.tsx

```diff
@@ -76,12 +78,15 @@ function ChartSettingsListColumns({
           _.isEqual(column.field_ref, fieldIdOrFieldRef),
       );
       if (column) {
-        onShowWidget({
-          id: "column_settings",
-          props: {
-            initialKey: keyForColumn(column),
+        onShowWidget(
+          {
+            id: "column_settings",
+            props: {
+              initialKey: keyForColumn(column),
+            },
           },
-        });
+          settingsRef.current,
```

Comment:
I think it might be possible to use a target from an event from `onClick` (line 125&144). This would target the correct element hopefully, as the change here only targets the last gear icon in the Left section, as there can be a single ref, but multiple gear icons (see image below).
![image](https://user-images.githubusercontent.com/1937582/187892991-3447d28b-77d8-4bb4-8298-c66d435ce979.png)


This would also help simplify the code quite a bit by being able to remove useRef and keep the code relatively the same.